service/src/main/java/uk/gov/mca/beacons/api/accountholder/application/MicrosoftGraphClient.java

```diff
@@ -0,0 +1,123 @@
+package uk.gov.mca.beacons.api.accountholder.application;
+
+import com.azure.identity.ClientSecretCredential;
+import com.azure.identity.ClientSecretCredentialBuilder;
+import com.microsoft.graph.authentication.TokenCredentialAuthProvider;
+import com.microsoft.graph.models.PasswordProfile;
+import com.microsoft.graph.models.User;
+import com.microsoft.graph.requests.GraphServiceClient;
+import java.util.List;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.mca.beacons.api.accountholder.domain.AccountHolder;
+import uk.gov.mca.beacons.api.configuration.MicrosoftGraphConfiguration;
+
+@Slf4j
+@Component("microsoftGraphClient")
+public class MicrosoftGraphClient implements AuthClient {
+
+  private final ClientSecretCredential clientSecretCredential;
+  private final TokenCredentialAuthProvider tokenCredAuthProvider;
+  private final List<String> scopes = List.of(
+    "https://graph.microsoft.com/.default"
+  );
+  private final GraphServiceClient graphClient;
+
+  @Autowired
+  public MicrosoftGraphClient(MicrosoftGraphConfiguration config) {
+    this.clientSecretCredential =
+      new ClientSecretCredentialBuilder()
+        .clientId(config.getClientId())
+        .clientSecret(config.getClientSecret())
+        .tenantId(config.getB2cTenantId())
+        .build();
+    this.tokenCredAuthProvider =
+      new TokenCredentialAuthProvider(scopes, clientSecretCredential);
+    this.graphClient =
+      GraphServiceClient
+        .builder()
+        .authenticationProvider(tokenCredAuthProvider)
+        .buildClient();
+  }
+
+  @Override
+  public uk.gov.mca.beacons.api.shared.domain.user.User createUser(
+    uk.gov.mca.beacons.api.shared.domain.user.User user
+  ) {
+    return null;
+  }
+
+  @Override
+  public uk.gov.mca.beacons.api.shared.domain.user.User createAzureAdUser(
+    AzureAdAccountHolder user
+  ) {
+    try {
+      User azAdUser = new User();
+      azAdUser.accountEnabled = true;
+      azAdUser.displayName = user.getFullName();
+      azAdUser.mail = user.getEmail();
+      azAdUser.mailNickname = user.getMailNickname();
+      azAdUser.userPrincipalName = user.getUserPrincipalName();
+
+      PasswordProfile passwordProfile = new PasswordProfile();
+      passwordProfile.forceChangePasswordNextSignIn = true;
+      passwordProfile.password = "xWwvJ]6NMw+bWH-d";
```

Comment:
👀 👀

Reply:
Forcing password change on next login is great, setting them to be a fixed password is not.

Reply:
Actually, forcing password update period is probably a bad idea. 

Steve has changed his first name to John. 

Notifies BRT, they change his first name. 

John goes to login and is now forced to change his password?

Leave PW stuff alone I think and they can do forgotten password workflow if they don't know the pw